src/Samples/Sarif.WorkItems.Sample/Program.cs

```diff
@@ -5,112 +5,83 @@
 
-            // IMPORTANT! There's no facility defined yet for creating/configuring titles, area paths, etc.
+            // 5. Now we have a set of SARIF files, each of which will drive creation of a single bug
+            IEnumerable<WorkItemFilingMetadata> workItemsToFile = SarifLog.CreateWorkItemFilingMetadataFromLogs(sarifLogsToFile);
 
-            try
-            {
-                IEnumerable<ResultGroup> resultGroups = workItemFiler.FileWorkItems(logFilePath).Result;
-                foreach (ResultGroup resultGroup in resultGroups)
-                {
-                    foreach (Result result in resultGroup.Results)
-                    {
-                        bool matched = guidToMatchedResultMap.TryGetValue(result.Guid, out Result originalResult);
-                        Debug.Assert(matched);
-                        originalResult.WorkItemUris = result.WorkItemUris;
-                    }
-                }
-            }
-            finally
+            AdoWorkItemFiler adoWorkItemFiler = new AdoWorkItemFiler()
             {
-                if (File.Exists(logFilePath)) File.Delete(logFilePath);
-            }
-        }
-
-        #region Stubs
-        private static SarifLog RetrieveCurrentSarif()
-        {
-            string sarifText = GetResourceText("Sarif.Sdk.Sample.SampleTestFiles.Current.sarif");
-            return JsonConvert.DeserializeObject<SarifLog>(sarifText);
-        }
+                FileAttachments = true
+            };
 
-        private static SarifLog RetrieveBaselineSarif()
-        {
-            string[] resourceNames = typeof(Program).Assembly.GetManifestResourceNames();
-            Console.WriteLine(String.Join("\r\n", resourceNames));
+            // 6. Wire up an event handler that gets called when the bug is filed
+            //    This callback will retrieve the 
+            adoWorkItemFiler.WorkItemFiledOrUpdated += (sender, eventArgs) =>
+            {
+                Dictionary<string, Result> guidToResultMap = guidMappingVisitor.GuidToResultMap;
+                WorkItemFilingMetadata metadata = eventArgs.Metadata;
 
-            string sarifText = GetResourceText("Sarif.Sdk.Sample.SampleTestFiles.Baseline.sarif");
-            return JsonConvert.DeserializeObject<SarifLog>(sarifText);
-        }
+                Debug.Assert(eventArgs.Metadata.Attachments.Count == 1);
+                var filedSarifLog = (SarifLog)eventArgs.Metadata.Attachments[0];
 
-        public static string GetResourceText(string resourceName)
-        {
-            string text = null;
+                Debug.Assert(filedSarifLog.Runs.Count == 1);
+                foreach (Result result in filedSarifLog.Runs[0].Results)
+                {
+                    bool matched = guidToResultMap.TryGetValue(result.Guid, out Result originalResult);
+                    Debug.Assert(matched);
+                    originalResult.WorkItemUris = new Uri[] { metadata.Uri };
```

Comment:
>originalResult [](start = 20, length = 14)

It's worth a comment that this is reference-equal to a result in the `matchedLog`, so when you write it out below, the work item URIs go along for the ride.

Reply:
Yes. Don't you have a comment, however, on the usability of this pattern? i.e., our machinery requires the user to handle the association between the original and split SARIF log files. this is a design weakness, in my opinion.

---
In reply to: [307088666](https://github.com/microsoft/sarif-sdk/pull/1589#discussion_r307088666) [](ancestors = 307088666)

Reply:
Well yes, I suppose if the user's happy path code requires a comment then it's not as usable as it should be.

---
In reply to: [307391618](https://github.com/microsoft/sarif-sdk/pull/1589#discussion_r307391618) [](ancestors = 307391618,307088666)